src/UniswapV4Migrator.sol

```diff
@@ -0,0 +1,267 @@
+                    uint8(Actions.MINT_POSITION),
+                    uint8(Actions.MINT_POSITION),
+                    uint8(Actions.SETTLE_PAIR),
+                    uint8(Actions.SWEEP)
+                );
+            }
+        } else {
+            if (isNoOpGovernance) {
+                params = new bytes[](2);
+                balance0 = ERC20(token0).balanceOf(address(this));
+                actions = abi.encodePacked(uint8(Actions.MINT_POSITION), uint8(Actions.SETTLE_PAIR));
+            } else {
+                params = new bytes[](3);
+                balance0 = ERC20(token0).balanceOf(address(this));
+                actions = abi.encodePacked(
+                    uint8(Actions.MINT_POSITION), uint8(Actions.MINT_POSITION), uint8(Actions.SETTLE_PAIR)
+                );
+            }
+        }
+
+        int24 lowerTick =
+            ((TickMath.MIN_TICK + 1) - poolKey.tickSpacing + 1) / poolKey.tickSpacing * poolKey.tickSpacing;
+        int24 upperTick = (TickMath.MAX_TICK - 1) / poolKey.tickSpacing * poolKey.tickSpacing;
+
+        int24 currentTick = TickMath.getTickAtSqrtPrice(sqrtPriceX96);
+
+        if (currentTick < lowerTick || currentTick > upperTick) revert TickOutOfRange();
+
+        liquidity = LiquidityAmounts.getLiquidityForAmounts(
+            sqrtPriceX96,
+            TickMath.getSqrtPriceAtTick(lowerTick),
+            TickMath.getSqrtPriceAtTick(upperTick),
+            uint128(balance0),
+            uint128(balance1)
+        );
+
+        if (liquidity == 0) revert ZeroLiquidity();
+
+        if (isNoOpGovernance) {
+            // For no-op governance, send all liquidity to the protocol locker
+            params[0] = abi.encode(
+                poolKey,
+                lowerTick,
+                upperTick,
+                uint128(liquidity),
+                uint128(balance0),
+                uint128(balance1),
+                address(this),
+                new bytes(0)
+            );
+
+            params[1] = abi.encode(poolKey.currency0, poolKey.currency1);
+        } else {
+            // Standard case: split liquidity 10/90
+            uint256 lockedLiquidity = liquidity / 10;
+            uint256 timeLockLiquidity = liquidity - lockedLiquidity;
+
+            // Liquidity for the protocol locker
+            params[0] = abi.encode(
+                poolKey,
+                lowerTick,
+                upperTick,
+                uint128(lockedLiquidity),
+                uint128(balance0) / 10,
+                uint128(balance1) / 10,
+                address(this),
+                new bytes(0)
+            );
+
+            // Liquidity for the Timelock
+            params[1] = abi.encode(
+                poolKey,
+                lowerTick,
+                upperTick,
+                uint128(timeLockLiquidity),
+                uint128(balance0),
+                uint128(balance1),
+                recipient,
+                new bytes(0)
+            );
```

Comment:
I'm concerned about how we're dividing the liquidity amounts and balance amounts separately. In case `lockedLiquidity` and/or `timeLockLiquidity` divides cleanly but `balance0/1` division requires rounding then we may not be providing a high enough `amount0/1Max` to provide the full `liquidity` amount. Still need to think of the cleanest way to do this but we'll want to underestimate the liquidity amounts a bit somehow

This is also another argument for why fuzzing is really important here